osu.Game/Screens/Play/SpectatorResultsScreen.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Screens;
+using osu.Game.Online.Spectator;
+using osu.Game.Scoring;
+using osu.Game.Screens.Ranking;
+
+namespace osu.Game.Screens.Play
+{
+    public class SpectatorResultsScreen : SoloResultsScreen
+    {
+        public SpectatorResultsScreen(ScoreInfo score)
+            : base(score)
+        {
+        }
+
+        [Resolved]
+        private SpectatorStreamingClient spectatorStreaming { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            spectatorStreaming.OnUserBeganPlaying += userBeganPlaying;
+        }
+
+        private void userBeganPlaying(int userId, SpectatorState state)
+        {
+            if (userId == Score.UserID)
+            {
+                Schedule(() =>
+                {
+                    if (this.IsCurrentScreen()) this.Exit();
+                });
+            }
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (spectatorStreaming != null)
+                spectatorStreaming.OnUserBeganPlaying -= userBeganPlaying;
```

Comment:
Wondering if worth duplicating this in an `OnExiting()` handler - not sure what guarantees we have wrt when this will get called when the stack is cleaning up exited screens.

Reply:
I think it should be fine due to the `Schedule` call guarding against execution